monai/apps/detection/transforms/dictionary.py

```diff
@@ -1326,7 +1328,11 @@ def __init__(
         super().__init__(self.image_keys + self.box_keys, prob, max_k, spatial_axes, allow_missing_keys)
         self.box_ref_image_keys = ensure_tuple_rep(box_ref_image_keys, len(self.box_keys))
 
-    def __call__(self, data: Mapping[Hashable, torch.Tensor]) -> Mapping[Hashable, torch.Tensor]:
+    def __call__(
+        self, data: Mapping[Hashable, torch.Tensor], lazy: bool | None = None
+    ) -> Mapping[Hashable, torch.Tensor]:
+        if lazy is True:
```

Comment:
```suggestion
        if lazy:
```

Reply:
I prefer `lazy is True` because lazy could be passed some arbitrary value by the user such as '123'

Reply:
and in that case, there's no warning at all, I don't think that's the expected behaviour of this input check.

Reply:
The difference is that if you accidentally pass a bad type to lazy, it is going to perform lazy evaluation. With `lazy is True`, you are only going to perform lazy evaluation if you pass it True. Given that this is an experimental feature, that feels safer to me